From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Snubby <87741849+SnubbyOWO@users.noreply.github.com>
Date: Fri, 22 May 2026 16:52:28 -0400
Subject: [PATCH] Add TOTEM_OF_UNDYING material alias


diff --git a/src/main/java/org/bukkit/Material.java b/src/main/java/org/bukkit/Material.java
index 91e89d8ec9cdc8e7e4afb63b32c7b2184f982935..69ea2e1b662e56c19aed5576f048b71b77c38639 100644
--- a/src/main/java/org/bukkit/Material.java
+++ b/src/main/java/org/bukkit/Material.java
@@ -522,6 +522,8 @@ public enum Material {
     BOAT_JUNGLE(446, 1),
     BOAT_ACACIA(447, 1),
     BOAT_DARK_OAK(448, 1),
+    @Deprecated
+    TOTEM_OF_UNDYING(449, 1),
     TOTEM(449, 1),
     SHULKER_SHELL(450),
     IRON_NUGGET(452),
diff --git a/src/test/java/org/bukkit/MaterialTest.java b/src/test/java/org/bukkit/MaterialTest.java
index 61c6871180da87f3d88a4f4f3fb3d2e838f421aa..3e9706b1d1ed22ad2962877f1cd73e03901b5bb5 100644
--- a/src/test/java/org/bukkit/MaterialTest.java
+++ b/src/test/java/org/bukkit/MaterialTest.java
@@ -82,4 +82,12 @@ public class MaterialTest {
             assertThat(Material.matchMaterial(name), is(material));
         }
     }
+
+    @Test
+    public void totemOfUndyingAlias() {
+        assertThat(Material.TOTEM_OF_UNDYING.getId(), is(Material.TOTEM.getId()));
+        assertThat(Material.getMaterial("TOTEM_OF_UNDYING"), is(Material.TOTEM_OF_UNDYING));
+        assertThat(Material.matchMaterial("totem of undying"), is(Material.TOTEM_OF_UNDYING));
+        assertThat(Material.getMaterial(Material.TOTEM.getId()), is(Material.TOTEM));
+    }
 }
-- 
2.40.0.windows.1

