From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Snubby <87741849+SnubbyOWO@users.noreply.github.com>
Date: Sat, 4 Jul 2026 17:24:00 -0400
Subject: [PATCH] another fix for via


diff --git a/src/main/java/net/minecraft/server/WorldServer.java b/src/main/java/net/minecraft/server/WorldServer.java
index 359814b145475b4c67a71c8c13924f70e9f3a1b6..63626d4d08f366427c8bb6a4a01c49dd5f8ad5ab 100644
--- a/src/main/java/net/minecraft/server/WorldServer.java
+++ b/src/main/java/net/minecraft/server/WorldServer.java
@@ -1257,9 +1257,31 @@ public class WorldServer extends World implements IAsyncTaskHandler {
     }
 
     public void broadcastEntityEffect(Entity entity, byte b0) {
+        // Beast start - FUCK YOUUUUUUUUUUUUUUUU
+        if (istheFUCKINGSHITDAMAGE(b0)) {
+            this.getTracker().sendPacketToEntity(entity, new PacketPlayOutAnimation(entity, 1));
+            return;
+        }
+        // Beast end
         this.getTracker().sendPacketToEntity(entity, new PacketPlayOutEntityStatus(entity, b0));
     }
 
+    // Beast start - fuck ass shit
+    private static boolean istheFUCKINGSHITDAMAGE(byte status) {
+        switch (status) {
+            case 2:
+            case 33:
+            case 36:
+            case 37:
+            case 44:
+            case 57:
+                return true;
+            default:
+                return false;
+        }
+    }
+    // Beast end
+
     public ChunkProviderServer getChunkProviderServer() {
         return (ChunkProviderServer) super.getChunkProvider();
     }
